osu.Game.Rulesets.Osu/Difficulty/OsuPerformanceCalculator.cs

```diff
@@ -95,11 +95,8 @@ private double computeAimValue()
                                  (totalHits > 2000 ? Math.Log10(totalHits / 2000.0) * 0.5 : 0.0);
             aimValue *= lengthBonus;
 
-            // Penalize misses by assessing # of misses relative to the total # of objects. Default a 3% reduction for any # of misses.
             if (effectiveMissCount > 0)
-                aimValue *= 0.97 * Math.Pow(1 - Math.Pow((double)effectiveMissCount / totalHits, 0.775), effectiveMissCount);
-
-            aimValue *= getComboScalingFactor();
+                aimValue *= calculateMissPenalty(effectiveMissCount, Attributes.AimDifficultStrainCount);
```

Comment:
You should calculate miss penalty only once and save it into a varible in my opinion.

Reply:
the penalties for aim and speed are different because `AimDifficultStrainCount` and `SpeedDifficultStrainCount` are different, so there is no repetition...